### PR TITLE
Move to single Activity architecture

### PR DESCRIPTION
**Pull request by braczkow**
As discussed: I have refactored the devstarter to a single-activity architecture.
One discussable point: I have removed BaseActivity (as we have only one either way!). 

If anyone has a problem with this, I can revert this change.

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -1,36 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<FrameLayout
```
Comment (krzdabrowski) [inline app/src/main/res/layout/activity_main.xml:2]: You could also consider [FragmentContainerView](https://developer.android.com/reference/androidx/fragment/app/FragmentContainerView) :)

File: app/build.gradle [hunk 2]
```diff
@@ -83,6 +83,7 @@ dependencies {
     implementation "androidx.navigation:navigation-ui-ktx:${versions.navigation}"
 
     implementation "androidx.lifecycle:lifecycle-reactivestreams:${versions.lifecycle}"
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (krzdabrowski) [inline app/build.gradle:86]: Is it really necessary? I've never really used that and thought that `androidx.appcompat:appcompat` includes all necessary "legacy" classes.

**Review by krzdabrowski — APPROVED** [review]

**Review by krzdabrowski — APPROVED** [review]

**Review by MDikkii — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 2 above)
Comment (braczkow) [reply app/build.gradle]: Removed, closing.

File: app/src/main/java/com/example/appName/data/repository/exampleuser/LocalExampleUserRepository.kt [hunk 3]
```diff
@@ -1,12 +1,15 @@
 package com.example.appName.data.repository.exampleuser
 
 import com.example.appName.data.model.ExampleUser
+import com.example.appName.presentation.features.base.SchedulersFactory
 import io.reactivex.rxjava3.core.Single
 import io.reactivex.rxjava3.schedulers.Schedulers
```
[the remark this answers is not in the document]
Comment (braczkow) [reply app/src/main/java/com/example/appName/data/repository/exampleuser/LocalExampleUserRepository.kt:6]: done

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -3,4 +3,6 @@
     <string name="action_login">Login</string>
     <string name="action_logout">Logout</string>
     <string name="main_welcome_text">Welcome %s!</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
[the remark this answers is not in the document]
Comment (braczkow) [reply app/src/main/res/values/strings.xml:7]: done

File: app/src/main/java/com/example/appName/presentation/features/main/MainFragment.kt [hunk 5]
```diff
@@ -41,5 +42,4 @@ class MainActivity : BaseActivity<MainViewState, MainPresenter>(
             logout_button.visibility = View.GONE
         }
     }
-    //endregion
-}
+}
```
[the remark this answers is not in the document]
Comment (braczkow) [reply app/src/main/java/com/example/appName/presentation/features/main/MainFragment.kt:45]: done

File: app/src/test/java/com/example/appName/utils/TestSchedulersFactory.kt [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.example.appName.presentation.features.base.SchedulersFactory
+import io.reactivex.rxjava3.core.Scheduler
+import io.reactivex.rxjava3.schedulers.Schedulers
+
+class TestSchedulersFactory(
+        override val io: Scheduler = Schedulers.trampoline(),
+        override val main: Scheduler = Schedulers.trampoline(),
+        override val reducer: Scheduler = Schedulers.trampoline()) : SchedulersFactory {
+}
```
[the remark this answers is not in the document]
Comment (braczkow) [reply app/src/test/java/com/example/appName/utils/TestSchedulersFactory.kt]: done

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (braczkow) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/build.gradle (hunk 2 above)
Comment (braczkow) [reply app/build.gradle]: Removed, done.

File: app/src/main/java/com/example/appName/data/repository/exampleuser/LocalExampleUserRepository.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (braczkow) [reply app/src/main/java/com/example/appName/data/repository/exampleuser/LocalExampleUserRepository.kt]: done

File: app/src/test/java/com/example/appName/presentation/features/main/MainPresenterTest.kt [hunk 7]
```diff
@@ -15,6 +18,11 @@ class MainPresenterTest : BaseTest() {
 
     private val testUser = ExampleUser("John")
 
+    @Before
+    fun setup() {
+        SchedulersFactory.set(TestSchedulersFactory())
+    }
+
```
[the remark this answers is not in the document]
Comment (braczkow) [reply app/src/test/java/com/example/appName/presentation/features/main/MainPresenterTest.kt:25]: I'm planning some UT-refinement task, so I'd leave it for now

**Review by MDikkii — APPROVED** [review]

**Pull request merged**